src/cider/nrepl.clj

```diff
@@ -435,13 +435,13 @@ Depending on the type of the return value of the evaluation this middleware may
            cause and stack frame info for the most recent exception."
     :requires #{#'session #'wrap-print}
     :expects #{}
-    :handles {"stacktrace" {:doc "Return messages describing each cause and stack frame of the most recent exception."
+    :handles {"analyze-stacktrace" {:doc "Parse the `stacktrace` and return messages describing each cause and stack frame."
+                                    :requires {"stacktrace" "The stacktrace to be parsed and analyzed as a string."}
+                                    :optional wrap-print-optional-arguments
+                                    :returns {"status" "\"done\", or \"no-error\" if `stracktrace` is not recognized"}}
+              "stacktrace" {:doc "Return messages describing each cause and stack frame of the most recent exception."
```

Comment:
> I think I like analyze-last-stacktrace but how would you do this alias? Just adding a new op and pointing to the same implementation? How do we deprecate the old one?

Yeah, just a new pointing to the same implementation. At some point it might be nice to add better support for this to nREPL itself. As for the deprecation - some note in the docs will do. You can also use `notify-client` to send them deprecation messages.

Reply:
@bbatsov I added the `analyze-last-stacktrace` op and made the `stacktrace` op deprecated. The deprecation is reflected in the docs and the `stacktrace` op also does send a notification to the user. I am not sure if this is a good idea, because I had to make a change in Cider to handle this notification in the response. See here:
https://github.com/clojure-emacs/cider/pull/3247/commits/0d0633aaac879d7ceef85a2546d74464321c7a44#diff-e5e7f12a3638dc838dbd4f61a5bd797efc0ac6120e58c80ec1a92970eea3a27cR479

I don't know which clients we have, but if they don't handle notification messages in a stacktrace response they might break. So I would suggest removing the client notification to not break anyone and just mark the stacktrace op deprecated in the docs. WDYT?

Reply:
Fair point about some clients probably not supporting this (I'm surprised we forgot to add it to CIDER :D ), so I guess we can just rely on a deprecation in the docs.